cdap-api/src/main/java/co/cask/cdap/api/lineage/LineageGraph.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Graph representing set of {@link Read} operations from the data sources,
+ * set of {@link Transform} operations, and set of {@link Write} operations to the data sinks.
+ */
+public class LineageGraph {
```

Comment:
I don't think this is the best name. It contains reads, writes and operations, no nodes or edges.

Reply:
Renamed to `Operations`. Let me know what do you think of it.